### PR TITLE
fix: use error object for response error

### PR DESCRIPTION
**Pull request by cprivitere**
We think we were losing the error message on line 62 due to not using an error object. This switches it to be more in line with 118 where we use an error object.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -223,6 +225,7 @@ async function run() {
     );
   } catch (error) {
     core.setFailed(`Failed to create server: ${error.message}`);
+    throw error;
```
Comment (ctreatma) [inline index.js:228]: I think this is the highest-level `try`/`catch` block, so we shouldn't re-throw here.  This is where we want to mark the action failed and let the process exit on its own, and I believe we've seen that work as expected when the top-level timeout is hit.

File: index.js [hunk 2]
```diff
@@ -178,6 +179,7 @@ async function getIPAddress(serverId) {
     return ipAddress;
   } catch (error) {
     core.setFailed(`Failed to get IP Address: ${error.message}`);
```
Comment (ctreatma) [inline index.js:181]: I think we should change these other `setFailed` calls to plain old logs

File: index.js [hunk 3]
```diff
@@ -132,6 +132,7 @@ async function getStatus(serverId) {
     return serverStatus;
   } catch (error) {
     core.setFailed(`Server provisioning failed: ${error.message}`);
+    return error;
```
Comment (ctreatma) [inline index.js:135]: This should throw, not return

File: index.js (hunk 2 above)
Comment (cprivitere) [reply index.js:181]: My learning has been that you don't wnat to use console functions in github actions. Instead, I'll switch them to core.info, which I believe gets all the things you want, without the console writing issues.

Comment (cprivitere) [reply index.js:181]: sorry, i'll set it to core.error

**Review by ctreatma — APPROVED** [review]
Ran a test with this branch and it failed fast!  💥

**Pull request merged**